contracts/FunctionsConsumer.sol

```diff
@@ -1,86 +1,80 @@
+   * @param encryptedSecretsReference Reference pointing to encrypted secrets
+   * @param args String arguments that will be passed into the source code
+   * @param bytesArgs Bytes arguments that will be passed into the source code as hex strings
+   * @param subscriptionId Subscription ID used to pay for request (FunctionsConsumer contract address must first be added to the subscription)
+   * @param callbackGasLimit Maximum amount of gas used to call the client contract's `handleOracleFulfillment` method
    */
-  function executeRequest(
+  function sendRequest(
     string calldata source,
-    bytes calldata secrets,
+    FunctionsRequest.Location secretsLocation,
+    bytes calldata encryptedSecretsReference,
     string[] calldata args,
+    bytes[] calldata bytesArgs,
     uint64 subscriptionId,
-    uint32 gasLimit
-  ) public onlyOwner returns (bytes32) {
-    Functions.Request memory req;
-    req.initializeRequest(Functions.Location.Inline, Functions.CodeLanguage.JavaScript, source);
-    if (secrets.length > 0) {
-      req.addRemoteSecrets(secrets);
+    uint32 callbackGasLimit
+  ) external onlyOwner {
+    FunctionsRequest.Request memory req;
+    req.initializeRequest(FunctionsRequest.Location.Inline, FunctionsRequest.CodeLanguage.JavaScript, source);
+    req.secretsLocation = secretsLocation;
+    req.encryptedSecretsReference = encryptedSecretsReference;
+    if (args.length > 0) {
+      req.setArgs(args);
     }
-    if (args.length > 0) req.addArgs(args);
-
-    bytes32 assignedReqID = sendRequest(req, subscriptionId, gasLimit);
-    latestRequestId = assignedReqID;
-    return assignedReqID;
+    if (bytesArgs.length > 0) {
+      req.setBytesArgs(bytesArgs);
+    }
+    s_lastRequestId = _sendRequest(req.encodeCBOR(), subscriptionId, callbackGasLimit, donId);
   }
 
   /**
-   * @notice Callback that is invoked once the DON has resolved the request or hit an error
-   *
+   * @notice Store latest result/error
    * @param requestId The request ID, returned by sendRequest()
    * @param response Aggregated response from the user code
    * @param err Aggregated error from the user code or from the execution pipeline
    * Either response or error parameter will be set, but never both
    */
   function fulfillRequest(bytes32 requestId, bytes memory response, bytes memory err) internal override {
-    latestResponse = response;
-    latestError = err;
-    emit OCRResponse(requestId, response, err);
-  }
-
-  /**
-   * @notice Allows the Functions oracle address to be updated
-   *
-   * @param oracle New oracle address
-   */
-  function updateOracleAddress(address oracle) public onlyOwner {
-    setOracle(oracle);
-  }
-
-  function addSimulatedRequestId(address oracleAddress, bytes32 requestId) public onlyOwner {
-    addExternalRequest(oracleAddress, requestId);
+    if (s_lastRequestId != requestId) {
+      revert UnexpectedRequestID(requestId);
```

Comment:
in what circumstances could this happen? how can we help end users debug this?

@aelmanaa do we have a section in the docs for common "gotchas" and debugging steps?

Reply:
This only applies if 2 requests are in-flight.  Then this contract will only allow for receiving the latest one.  However, we could just remove this to allow for multiple in-flight responses.

I am going to remove the `if` statement since it is not necessary and just adds complexity.